ferrocene/ci/scripts/mirror-sphinx-shared-resources.py

```diff
@@ -27,40 +29,63 @@ def main():
 
     # As an input the script takes two environment variables. They specify the
     # (relative) path to the origin and mirror repository.
-    origin_repo_path = os.environ["ORIGIN_REPO_PATH"]
-    mirror_repo_path = os.environ["MIRROR_REPO_PATH"]
-    if origin_repo_path == "" or mirror_repo_path == "":
+    origin_repo_path = os.getenv("ORIGIN_REPO_PATH")
+    mirror_repo_path = os.getenv("MIRROR_REPO_PATH")
+    if origin_repo_path == None or mirror_repo_path == None:
         raise Exception("script needs env vars ORIGIN_REPO_PATH and MIRROR_REPO_PATH")
 
+    last_mirrored_commit = get_last_mirrored_commit(mirror_repo_path)
+
     # create patch from the origin repo and map the paths from `ORIGIN_DIR` to
     # `MIRROR_DIR`. We need to map it in order to apply it in the other repo.
     patch = run(
-        ["git", "diff", "HEAD~1", "HEAD", "--", ORIGIN_DIR], origin_repo_path
+        ["git", "diff", last_mirrored_commit, "HEAD", "--", ORIGIN_DIR],
+        origin_repo_path,
     ).stdout
     if patch == "":
         raise Exception(
-            f"patch cannot be empty; make sure the last commit made changes to \
-            {ORIGIN_DIR} (the ORIGIN_DIR)"
+            "patch cannot be empty; make sure the last commit made changes to "
+            + ORIGIN_DIR
         )
     patch = patch.replace(ORIGIN_DIR, MIRROR_DIR)
 
     # apply patch to the mirror repo and create a commit
     run(["git", "apply", "-"], mirror_repo_path, input=patch)
     run(["git", "add", "."], mirror_repo_path)
-    run(["git", "commit", "-m", COMMIT_MSG], mirror_repo_path)
+    commit_msg = get_commit_msg(origin_repo_path)
+    run(["git", "commit", "-m", commit_msg], mirror_repo_path)
+
+
+def get_commit_msg(origin_repo_path: str) -> str:
+    # get the hash of the commit which triggered the workflow ...
+    origin_commit_hash = run(["git", "rev-parse", "HEAD"], origin_repo_path).stdout
+    # ... and construct the commit message with it
+    return COMMIT_MSG_START + origin_commit_hash
+
+
+def get_last_mirrored_commit(mirror_repo_path: str) -> str:
+    commit_message = run(
+        ["git", "log", "--format=%B", "-n", "1", "HEAD"], mirror_repo_path
```

Comment:
```suggestion
        ["git", "log", "--format=%B", "-n", "1", "HEAD"], mirror_repo_path
```

Checking only the first commit (`-n 1`) will break if we send PRs to the mirrored repo only. While *in general* we shouldn't be doing it, I can see us maintaining a separate README or license files in the mirrored repo.

I would remove the `-n 1` and get concatenated commit bodies of all the commits in the history. Then we can iterate line by line and break at the first line beginning with `mirrored-commit: `.

Reply:
Changed it